finality-aleph/src/network/service.rs

```diff
@@ -393,11 +594,23 @@ mod tests {
         }
     }
 
+    fn message(i: u8) -> NetworkData<MockData, LegacyMockMultiaddress> {
+        NetworkData::Data(vec![i, i + 1, i + 2], SessionId(1))
+    }
+
+    /// This is a dummy implementation so that VersionedNetworkData can be put in HashSet.
+    /// It will inserted with other data, so this can always return the same thing.
```

Comment:
In the tests it is sometime inserted with more data. And we need HashSet, as we do not have control over order of peers during broadcast. (technically we could order expected messages so that the test pass,  but I would call that a bad idea)

Reply:
Ok, nevermind. I derived hash for those as described in https://www.notion.so/cardinalcryptography/Basic-traits-derivation-and-field-access-for-public-structs-0aa22c4887024547b613db5754e0664a